snuba/datasets/storages/processors/replaced_groups.py

```diff
@@ -43,40 +51,72 @@ def process_query(self, query: Query, request_settings: RequestSettings) -> None
 
         set_final = False
         if project_ids:
-            final, exclude_group_ids, replacement_types = get_projects_query_flags(
-                list(project_ids), self.__replacer_state_name,
-            )
+            (
+                query_is_final,
+                exclude_group_ids,
+                replacement_types,
+            ) = get_projects_query_flags(list(project_ids), self.__replacer_state_name,)
             tags = {replacement_type: "True" for replacement_type in replacement_types}
             tags["referrer"] = request_settings.referrer
             tags["parent_api"] = request_settings.get_parent_api()
-            if final:
+            if query_is_final:
                 tags["cause"] = "final_flag"
+
                 metrics.increment(
                     "final", tags=tags,
                 )
-            if not final and exclude_group_ids:
-                # If the number of groups to exclude exceeds our limit, the query
-                # should just use final instead of the exclusion set.
-                max_group_ids_exclude = get_config(
-                    "max_group_ids_exclude", settings.REPLACER_MAX_GROUP_IDS_TO_EXCLUDE
-                )
-                assert isinstance(max_group_ids_exclude, int)
-                if len(exclude_group_ids) > max_group_ids_exclude:
-                    tags["cause"] = "max_groups"
-                    metrics.increment(
-                        "final", tags=tags,
+            if not query_is_final and exclude_group_ids:
+                if self._query_overlaps_replacements(query, project_ids):
```

Comment:
Please invert the condition, the code will be less nested and easier to read sequentially. 
```
if not self._query_overlaps_replacements(...):
    return
```

Reply:
Since that last line of the function still needs to run, it would look like:
```
if not self._query_overlaps_replacements(...):
    query.set_from_clause(replace(query.get_from_clause(), final=False))
    return
...
```
It'll reduce 1 nesting level but it means that `query.set_from_clause()` would need to be duplicated there.